pkg/middleware/auth_proxy/auth_proxy.go

```diff
@@ -168,57 +169,62 @@ func (auth *AuthProxy) getKey() string {
 	return fmt.Sprintf(CachePrefix, hashedKey)
 }
 
-// Login logs in user id with whatever means possible
-func (auth *AuthProxy) Login() (int64, *Error) {
-
-	id, _ := auth.GetUserViaCache()
-	if id != 0 {
+// Login logs in user ID by whatever means possible.
+func (auth *AuthProxy) Login(logger log.Logger, ignoreCache bool) (int64, *Error) {
+	if !ignoreCache {
 		// Error here means absent cache - we don't need to handle that
-		return id, nil
+		id, err := auth.GetUserViaCache(logger)
+		if err == nil && id != 0 {
+			return id, nil
+		}
 	}
 
 	if isLDAPEnabled() {
 		id, err := auth.LoginViaLDAP()
-
-		if err == ldap.ErrInvalidCredentials {
-			return 0, newError(
-				"Proxy authentication required",
-				ldap.ErrInvalidCredentials,
-			)
-		}
-
 		if err != nil {
-			return 0, newError("Failed to get the user", err)
+			if err == ldap.ErrInvalidCredentials {
+				return 0, newError("proxy authentication required", ldap.ErrInvalidCredentials)
+			}
+			return 0, newError("failed to get the user", err)
 		}
 
 		return id, nil
 	}
 
 	id, err := auth.LoginViaHeader()
 	if err != nil {
-		return 0, newError(
-			"Failed to log in as user, specified in auth proxy header",
-			err,
-		)
+		return 0, newError("failed to log in as user, specified in auth proxy header", err)
 	}
 
 	return id, nil
 }
 
-// GetUserViaCache gets user id from cache
-func (auth *AuthProxy) GetUserViaCache() (int64, error) {
-	var (
-		cacheKey    = auth.getKey()
-		userID, err = auth.store.Get(cacheKey)
-	)
-
+// GetUserViaCache gets user ID from cache.
+func (auth *AuthProxy) GetUserViaCache(logger log.Logger) (int64, error) {
+	cacheKey := auth.getKey()
+	logger.Debug("Getting user ID via auth cache", "cacheKey", cacheKey)
+	userID, err := auth.store.Get(cacheKey)
 	if err != nil {
+		logger.Debug("Failed getting user ID via auth cache", "error", err)
```

Comment:
@papagian It shouldn't be logged as an error, since it isn't something that needs to be dealt with (i.e., it's only technically an error, not logically). It will just be confusing noise in the logs for anyone looking for errors. See [here](https://github.com/grafana/grafana/pull/23979/files#diff-dc90ed542fb29715788afbc5a913c52cR175).

Reply:
I don't insist :)